# SPDX-License-Identifier: BSD-2-Clause
# Origin: EmberBSD contributors (AI-assisted).
# Rebuild boost-headers with the NetBSD wait status fix in
# boost/process/v2/exit_code.hpp, and build boost-libs against it.

--- a/devel/boost-headers/Makefile
+++ b/devel/boost-headers/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.76 2026/05/15 09:13:49 adam Exp $
 
 BOOST_PACKAGE=		headers
+PKGREVISION=		1
 BOOST_COMMENT=		(build-time headers)
 BOOST_CONFIG=		generate
 
--- a/devel/boost-libs/Makefile
+++ b/devel/boost-libs/Makefile
@@ -91,6 +91,8 @@
 .include "../../archivers/xz/buildlink3.mk"
 .include "../../archivers/zstd/buildlink3.mk"
 .include "../../converters/libiconv/buildlink3.mk"
+# boost/process/v2/exit_code.hpp builds on NetBSD from nb1.
+BUILDLINK_API_DEPENDS.boost-headers+=	boost-headers>=1.92.0nb1
 .include "../../devel/boost-headers/buildlink3.mk"
 .include "../../devel/zlib/buildlink3.mk"
 PKG_OPTIONS_VAR= PKG_OPTIONS.boost-libs
